deepchecks/utils/gower_distance.py

```diff
@@ -64,43 +67,78 @@ def gower_matrix_n_closets(data: np.ndarray, cat_features: np.array, num_neighbo
     In addition, it can deal with missing values.
     Parameters
     ----------
-    data: numpy.ndarray
-        Dataset matrix.
-    cat_features: numpy.array
-        Boolean array of representing which of the columns are categorical features.
+    cat_data: pd.DataFrame
+        The categorical features part of the dataset.
+    numeric_data: pd.DataFrame
+        The numerical features part of the dataset.
     num_neighbours: int
         Number of neighbours to return. For example, for n=2 for each sample returns the distances to the two closest
         samples in the dataset.
     Returns
     -------
     numpy.ndarray
-     representing the distance matrix to the nearest neighbours.
+        representing the distance matrix to the nearest neighbours.
     numpy.ndarray
-     representing the indexes of the nearest neighbours.
+        representing the indexes of the nearest neighbours.
     """
-    if not isinstance(data, np.ndarray):
-        data = np.asarray(data)
+    num_samples = cat_data.shape[0]
+    num_features = cat_data.shape[1] + numeric_data.shape[1]
+    distances, indexes = np.zeros((num_samples, num_neighbours)), np.zeros((num_samples, num_neighbours))
+    # handle categorical - transform to an ordinal numpy array
+    enc = OrdinalEncoder()
+    cat_data = np.nan_to_num(enc.fit_transform(cat_data.fillna(value=np.nan)), nan=-1)
```

Comment:
Does pandas fillna fill also pd.NaN, pd.NaT, and python None?

Reply:
Yes, can see in tests :)